test/e2e/test/elasticsearch/checks_es.go

```diff
@@ -349,6 +349,13 @@ func compareCgroupMemoryLimit(topologyElement esv1.NodeSet, nodeStats client.Nod
 		// no expected memory, consider it's ok
 		return nil
 	}
+
+	if len(nodeStats.OS.CGroup.Memory.LimitInBytes) == 0 {
+		// Elasticsearch versions before 7.16 cannot parse cgroup v2 information and
+		// will have no information in this field. Considering it ok.
+		return nil
+	}
```

Comment:
If I'm being paranoid, I'd say we should add a version gate, otherwise we might miss a regression in the future where a newer ES version might fail to parse the cgroup info?

Reply:
We can do that but have to look at the version for each pod, because we can have mixed versions during an upgrade where some nodes have the info and some not.